lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -852,10 +911,16 @@ export abstract class CacheManager implements ICacheManager {
         const cachedAppMetadata = this.readAppMetadataFromCache(environment);
 
         if (cachedAccount && cachedIdToken) {
-            cachedAccount.idTokenClaims = extractTokenClaims(
+            const idTokenClaims = extractTokenClaims(
                 cachedIdToken.secret,
                 this.cryptoImpl.base64Decode
             );
+
+            cachedAccount.idTokenClaims = idTokenClaims;
+            cachedAccount.localAccountId =
+                idTokenClaims.oid || cachedAccount.localAccountId;
+            cachedAccount.name = idTokenClaims.name || cachedAccount.name;
+            cachedAccount.realm = idTokenClaims.tid || cachedAccount.realm;
```

Comment:
Why are we overwriting the account properties here? In any case this logic feels weird to me - write contents from idToken or write property from self. 

Makes more sense to only overwrite if contents from idToken are truthy and != to existing contents IMO.

Reply:
Added a method that updates the tenant specific data from the claims